dna-src/zomes/my_zome/code/src/lib.rs

```diff
@@ -164,20 +164,44 @@ mod my_zome {
     }
 
     #[zome_fn("hc_public")]
-    fn create_goal(goal: Goal) -> ZomeApiResult<GetResponse<Goal>> {
+    fn create_goal(goal: Goal, parent_address: Option<Address>) -> ZomeApiResult<GetResponse<Goal>> {
         let app_entry = Entry::App("goal".into(), goal.clone().into());
-        let _ = hdk::commit_entry(&app_entry)?;
+        let entry_address = hdk::commit_entry(&app_entry)?;
 
-        // link each new goal to the anchor
+        // link new goal to the anchor
         let anchor_address = Entry::App(
             "anchor".into(), // app entry type
             "goals".into() // app entry value
         ).address();
 
         hdk::link_entries(&anchor_address, &app_entry.address(),  "anchor->goal", "")?;
 
+        // if a parent address was provided, link the goal with its parent
+        match parent_address {
```

Comment:
We could assign the result of this match call to a variable... like:
```suggestion
        let maybe_edge = match parent_address {
```